tpm/tpm.go

```diff
@@ -713,6 +714,106 @@ func OwnerReadSRK(rw io.ReadWriter, ownerAuth digest) ([]byte, error) {
 	return tpmutil.Pack(pk)
 }
 
+// ReadEKCert reads the EKCert from the NVRAM.
+// TCG PC Client Specific Implementation Specification for Conventional BIOS 7.4.4
+func ReadEKCert(rw io.ReadWriter, ownAuth digest) ([]byte, error) {
+	const (
+		certIndex                 = 0x1000f000
+		certTagPCClientStoredCert = 0x1001
+		certTagPCClientFullCert   = 0x1002
+	)
+	offset := uint32(0)
+	data, err := NVReadValue(rw, certIndex, offset, 5, ownAuth)
+	if err != nil {
+		return nil, err
+	}
+	offset = offset + 5
+	buff := bytes.NewReader(data)
+
+	var header struct {
+		Tag      uint16
+		CertType uint8
+		CertSize uint16
+	}
+	var bufSize uint32
```

Comment:
nit: From the placement of this declaration, I assumed it was needed in the next line, but it's not needed until the switch. Consider moving it down to just before the switch to make that clear.